lldb/packages/Python/lldbsuite/test/tools/lldb-dap/dap_server.py

```diff
@@ -27,6 +27,10 @@
     Literal,
 )
 
+# set timeout based on whether ASAN was enabled or not. Increase
+# timeout by a factor of 10 if ASAN is enabled.
+DEFAULT_TIMEOUT = 10 * (10 if ("ASAN_OPTIONS" in os.environ) else 1)
```

Comment:
There is one other case where we may want a longer timeout.

When we're using a `CMAKE_BUILD_TYPE=Debug` the tests do have more timeouts. I think we've tagged a few places, but we could also increase the timeout for those as well.

Reply:
Yeah, this probably warrants a centralized helper, maybe do 2x for debug builds, 10x for sanitized builds, etc.